### ISSUE TITLE
Infinite loop when parsing malformed/non-standard markdown

### ISSUE THREAD

**Issue by nickw444** [lepture/mistune#213]
Example code:
```python
import mistune

markdown = mistune.create_markdown(renderer=mistune.HTMLRenderer())
markdown('''
. Foo 
. Bar
. Baz
''')
```

It appears to be getting stuck in `scanner.py`:

https://github.com/lepture/mistune/blob/75884268396ed6b1cb846b610facebe284eee1f3/mistune/scanner.py#L106-L120

```
  File "mistune_test.py", line 8, in <module>
    ''')
  File "/Users/nickw/.local/share/virtualenvs/100-warm-tunas-pipeline-NrcgpDFS/lib/python3.7/site-packages/mistune/markdown.py", line 69, in __call__
    return self.parse(s)
  File "/Users/nickw/.local/share/virtualenvs/100-warm-tunas-pipeline-NrcgpDFS/lib/python3.7/site-packages/mistune/markdown.py", line 52, in parse
    tokens = self.block.parse(s, state)
  File "/Users/nickw/.local/share/virtualenvs/100-warm-tunas-pipeline-NrcgpDFS/lib/python3.7/site-packages/mistune/block_parser.py", line 240, in parse
    return list(self._scan(s, state, rules))
  File "/Users/nickw/.local/share/virtualenvs/100-warm-tunas-pipeline-NrcgpDFS/lib/python3.7/site-packages/mistune/scanner.py", line 56, in _scan
    for tok in sc.iter(s, state, self.parse_text):
  File "/Users/nickw/.local/share/virtualenvs/100-warm-tunas-pipeline-NrcgpDFS/lib/python3.7/site-packages/mistune/scanner.py", line 117, in iter
    token = method(match, state, string)
  File "/Users/nickw/.local/share/virtualenvs/100-warm-tunas-pipeline-NrcgpDFS/lib/python3.7/site-packages/mistune/block_parser.py", line 155, in parse_list_start
    items, pos = _find_list_items(string, m.start(), spaces, marker)
  File "/Users/nickw/.local/share/virtualenvs/100-warm-tunas-pipeline-NrcgpDFS/lib/python3.7/site-packages/mistune/block_parser.py", line 315, in _find_list_items
    pattern = _create_list_item_pattern(spaces, marker)
  File "/Users/nickw/.local/share/virtualenvs/100-warm-tunas-pipeline-NrcgpDFS/lib/python3.7/site-packages/mistune/block_parser.py", line 300, in _create_list_item_pattern
    r'(?:\1(?:' + s1 + '|' + s2 + ')'
  File "/Users/nickw/.local/share/virtualenvs/100-warm-tunas-pipeline-NrcgpDFS/lib/python3.7/re.py", line 232, in compile
    def compile(pattern, flags=0):
KeyboardInterrupt
```

**Comment by lepture** [comment]
Thanks for your reporting. Fixed in v2.0.0a2.

**Comment by nickw444** [comment]
Wow thanks for the speedy fix! This is a great library by the way 👍
